nucleus/payara-modules/notification-core/src/main/java/fish/payara/nucleus/notification/domain/NotifierExecutionOptionsFactoryStore.java

```diff
@@ -56,7 +57,7 @@
 public class NotifierExecutionOptionsFactoryStore {
 
     private Map<NotifierType, NotifierExecutionOptionsFactory> execOptionsFactoryStore =
-            new HashMap<NotifierType, NotifierExecutionOptionsFactory>();
+            new ConcurrentHashMap<NotifierType, NotifierExecutionOptionsFactory>();
```

Comment:
If I get HK2 semantics right this map could be accessed and modified by multiple threads so it should be thread safe.